### PR TITLE
=str #16935 revive dsl factories consistency spec

### LINKED ISSUE #16935: Revive DslConsistencySpec
Currently it is disabled, and it is not trivial to revive because of the arg 1-22 type methods on DSLs.

### PR DESCRIPTION
**Pull request by 2m**
Fixes #16935

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/DslFactoriesConsistencySpec.scala [hunk 1]
```diff
@@ -54,52 +52,69 @@ class DslFactoriesConsistencySpec extends WordSpec with Matchers {
...
-  def runSpec(sClass: Class[_], jClass: Class[_]) {
-    val jMethods = getJMethods(jClass)
-    val sMethods = getSMethods(sClass)
+  private val adaptCreate: PartialFunction[Method, Method] = {
+    case m if m.parameterTypes.size > 1 ⇒
+      // rename createN => create
+      // and adapt java side non curried functions to scala side like
+      m.copy(name = "create", parameterTypes = m.parameterTypes.dropRight(1) :+ classOf[akka.japi.function.Function[_, _]])
+    case m ⇒ m
+  }
```
Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/DslFactoriesConsistencySpec.scala]: :+1:

File: akka-stream-tests/src/test/scala/akka/stream/DslFactoriesConsistencySpec.scala [hunk 2]
```diff
@@ -54,52 +52,69 @@ class DslFactoriesConsistencySpec extends WordSpec with Matchers {
...
+  private def ignore(m: Method): Boolean = {
+    val ignores = Seq(
+      // private scaladsl method
+      Ignore(_ == akka.stream.scaladsl.Source.getClass, _ == "apply", _ == 1, _ == List(classOf[akka.stream.impl.SourceModule[_, _]])),
+      // corresponding matches on java side would need to have Function23
+      Ignore(_ == akka.stream.scaladsl.Source.getClass, _ == "apply", _ == 24, _ ⇒ true),
+      Ignore(_ == akka.stream.scaladsl.Flow.getClass, _ == "apply", _ == 24, _ ⇒ true),
+      Ignore(_ == akka.stream.scaladsl.Sink.getClass, _ == "apply", _ == 24, _ ⇒ true),
+      // all generated methods like scaladsl.Sink$.akka$stream$scaladsl$Sink$$newOnCompleteStage$1
+      Ignore(_ ⇒ true, _.contains("$"), _ ⇒ true, _ ⇒ true))
```
Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/DslFactoriesConsistencySpec.scala]: nice :)

**Comment by ktoso** [comment]
LGTM, awesome revival!

**Comment by drewhk** [comment]
LGTM.

We only miss the graph APIs now.

**Comment by ktoso** [comment]
Last commit LGTM as well :)

**Comment by 2m** [comment]
This now tests junction factories as well and completes this PR.

**Comment by ktoso** [comment]
LGTM once more :-)

**Pull request merged**